nisystemlink/clients/spec/models/_create_specs_request.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime
+from typing import List, Optional
+
+from nisystemlink.clients.core import ApiError
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+from nisystemlink.clients.spec.models._specification import SpecificationBase
+
+
+class CreateSpecificationRequestObject(SpecificationBase):
+    """Used to create specifications."""
+
+    pass
+
+
+class CreateSpecificationsRequest(JsonModel):
+    """Create multiple specifications."""
+
+    specs: Optional[List[CreateSpecificationRequestObject]]
+    """List of specifications to be created."""
+
+
+class CreateSpecificationResponseObject(JsonModel):
```

Comment:
I guess this can't extend from `SpecificationBase`? Looking at the Swagger doc, I do see that the created response doesn't include a lot of fields from the newly created spec.

There's a lot of redundancy here, though - I think we could do better since Python has multiple inheritance. 
Here's my quick analysis:
It looks like there's 3 different spec representations - the created/updated response, the query response, and then the update request.

All 3 models include `id, product_id, spec_id, version, workspace`, so that can be `SpecificationBase`.

Then there's the fields that are kind of like the "definition" of the spec, which are shared by the query response and update request: `block, category, conditions, keyword, limit, name, properties, symbol, type, unit`. So that can be a `SpecificationDefinition` type or whatever you want to call it.

So then the query response model could be something like:

```python
class Specification(SpecificationBase, SpecificationDefintion):
  created_at = ...
  created_by = ...
  updated_at = ...
  updated_by = ...
```
I guess we could extract the created/updated properties into separate types as well, but that's not a big deal.

I probably got something wrong above, but you get the idea.

Reply:
There was a bit more complexity than originally captured in the comment. I've now consolidated all of the core specification information into appropriate base classes. 
Turns out, that doing this exposed some oddities in the API definition that I'm going to want to ask for fixes on. But we'll update this API once the REST api is also updated.